packages/manager/modules/iam/src/iam.service.js

```diff
@@ -0,0 +1,429 @@
+   * Format the given list of guides based on the current user's subsidiary
+   * for direct use in templates
+   * @param {GUIDE[]} guides A list of GUIDE constants
+   * @returns {{ key: string, link: string, title: string, description: string }[]}
+   */
+  formatGuides(...guides) {
+    return Object.entries(GUIDE)
+      .filter(([, guide]) => guides.includes(guide))
+      .map(([key, guide]) => {
+        const lowerKey = key.toLowerCase();
+        return {
+          key: lowerKey,
+          link: guide[this.coreConfig.getUser().ovhSubsidiary] || guide.DEFAULT,
+          title: this.$translate.instant(`iam_guide_${lowerKey}`),
+          description: this.$translate.instant(
+            `iam_guide_description_${lowerKey}`,
+          ),
+          // TODO augment with tracking data
+        };
+      });
+  }
+
+  // **********************************************************************************************
+  // Identity
+
+  /**
+   * Get the list of identity groups
+   * @returns {Promise<string[]>}
+   */
+  getIdentityGroups() {
+    return this.$http.get(URL.IDENTITY_GROUP).then(({ data }) => data);
+  }
+
+  /**
+   * Get the list of identity users
+   * @returns {Promise<string[]>}
+   */
+  getIdentityUsers() {
+    return this.$http.get(URL.IDENTITY_USER).then(({ data }) => data);
+  }
+
+  // **********************************************************************************************
+  // Policy
+
+  /**
+   * Create a new policy
+   * @param {{
+   *   identities?: string[]
+   *   name: string
+   *   permissions: {
+   *     allow?: { action: string }[]
+   *     except?: { action: string }[]
+   *   }
+   *   resources?: { urn: string }[]
+   * }} data The policy's data
+   * @returns {Promise}
+   */
+  createPolicy(data) {
+    return this.httpApiv2({ method: 'post', url: URL.POLICY, data }).then(
+      ({ data: policy }) => policy,
+    );
+  }
+
+  /**
+   * Delete the policy given the id
+   * @param {string} id The policy's id
+   * @returns {Promise}
+   */
+  deletePolicy(id) {
+    return this.httpApiv2({
+      method: 'delete',
+      url: `${URL.POLICY}/${id}`,
+    }).then(({ data }) => data);
+  }
+
+  /**
+   * Get the policy given the id populated with more data
+   * @param {string} id The policy's id
+   * @returns {Promise}
+   */
+  getDetailedPolicy(id) {
+    return this.$q
+      .all({
+        actions: this.getActions(),
+        policy: this.httpApiv2({
+          method: 'get',
+          url: `${URL.POLICY}/${id}`,
+          params: { details: true },
+        }).then(({ data: policy }) => policy),
+      })
+      .then(({ actions, policy }) =>
+        IAMService.transformPolicy(policy, actions),
+      );
+  }
+
+  /**
+   * Get the list of policies
+   * @param {string} cursor The base64 encoded cursor to pass
+   * @param {boolean} readOnly
+   * @returns {Promise}
+   */
+  getPolicies({ cursor, readOnly }) {
+    const params = {};
+    if (typeof readOnly !== 'undefined') {
```

Comment:
**question:** Are you sure about checking `undefined` as a string?

Reply:
Did you miss the `typeof` operator ? :smiley:

Reply:
Yes you're right. I didn't see `typeof`